src/python/WMCore/ReqMgr/CherryPyThreads/StepChainParentageFixTask.py

```diff
@@ -68,23 +68,33 @@ def fixStepChainParentage(self, config):
         fixCount = 0
         for childDS in childDatasets:
             self.logger.info("Resolving parentage for dataset: %s", childDS)
-            start = int(time.time())
-            failedBlocks = self.dbsSvc.fixMissingParentageDatasets(childDS, insertFlag=True)
-            end = int(time.time())
-            timeTaken = end - start
-            if failedBlocks:
-                self.logger.warning("Failed to fix the parentage for %s will be retried: time took: %s (sec)",
-                                 failedBlocks, timeTaken)
+            start = time.time()
+            try:
+                failedBlocks = self.dbsSvc.fixMissingParentageDatasets(childDS, insertFlag=True)
+            except Exception as exc:
+                self.logger.exception("Failed to resolve parentage data for dataset: %s. Error: %s",
+                                      childDS, str(exc))
                 failedRequests = failedRequests.union(requestsByChildDataset[childDS])
             else:
-                fixCount += 1
-                self.logger.info("Fixed %s parentage: %s out of %s datasets. time took: %s (sec)",
-                                 childDS, fixCount, totalChildDS, timeTaken)
+                if failedBlocks:
+                    self.logger.warning("These blocks failed to be resolved and will be retried later: %s",
+                                        failedBlocks)
+                    failedRequests = failedRequests.union(requestsByChildDataset[childDS])
+                else:
+                    fixCount += 1
+                    self.logger.info("Parentage for '%s' successfully updated. Processed %s out of %s datasets.",
+                                     childDS, fixCount, totalChildDS)
+            timeTaken = time.time() - start
+            self.logger.info("    spent %s secs on this dataset: %s", timeTaken, childDS)
 
         requestsToUpdate = requests - failedRequests
 
         for request in requestsToUpdate:
-            self.reqmgrDB.updateRequestProperty(request, {"ParentageResolved": True})
-
-        self.logger.info("Total %s requests' ParentageResolved flag is set to True", len(requestsToUpdate))
-        self.logger.info("Total %s requests will be retried next cycle: %s", len(failedRequests), failedRequests)
+            try:
+                self.logger.info("Marking ParentageResolved=True for request: %s", request)
+                self.reqmgrDB.updateRequestProperty(request, {"ParentageResolved": True})
```

Comment:
Shouldn't those two be swapped and the logger message to say:
`self.logger.info("Marked ParentageResolved=True for request: %s", request)`

This way the log will not be polluted at all with messages regarding actions supposedly failing into the next line, while at the same time the relevant info will be provided by the `looging` message from the `except` clause anyway.

Reply:
I like putting logs before the action because the action might take a long time, and we know exactly where we are. But in this case, it indeed makes more sense to swap them. I'll update it.